daffodil-test/src/test/resources/org/apache/daffodil/layers/ipv4.dfdl.xsd

```diff
@@ -0,0 +1,261 @@
+
+          At parse time we must check that this computed checksum value, and the
+          actual value of the element in the infoset are equal.
+
+          We have an option to place the computed checksum in another element
+          and then have a schematron validation rule or just a DFDL assert (recoverable)
+          check that the checksum element and computed checksum are equal.
+
+          Below, that element is called "ComputedChecksum".
+
+          When Unparsing, the first part transform computes the partial checksum of
+          the part 1 data, placing the result of that into the checksumPart1Value
+          just as when parsing. This doesn't suspend itself. It just
+          waits until all bytes of the first part of the header are written. That
+          itself can involve all sorts of suspensions though, as the header fields in this
+          part contain the length of the IP packet data, which likely involves a forward
+          reference to the dfdl:valueLength of the payload.
+
+          The second part transform computes the final checksum combining the checksumPart1Value
+          variable with the checksum computed from the part 2 data, placing the result into
+          checksumValue variable. This suspends if the checksumPart1Value variable is not
+          yet available, and also requires that all bytes of the layer itself have been written.
+
+          A dfdl:outputValueCalc puts this newly computed value from the variable into the checksum element.
+          Any prior value of the checksum element is discarded/overwritten with the new value.
+
+          -->
+          <dfdl:newVariableInstance ref="chksum:ipV4ChecksumPart1Value"/>
+          <dfdl:newVariableInstance ref="chksum:ipV4ChecksumValue"/>
+        </xs:appinfo>
+      </xs:annotation>
+
+      <xs:sequence dfdl:ref="chksum:ipv4ChecksumLayerPart1">
+        <xs:sequence>
+        <!--
+        Within this sequence, the data is all included in Part 1 of the
+        ipV4 checksum calculation
+
+        On parsing, the checksum algorithm computes the partial checksum for
+        this first part of the data, placing the value into the
+        variable chksum:checksumPart1Value, which must have a new variable
+        instance in scope.
+
+        This checksum calculation knows exactly how big this data must be
+        and if the representation turned out to be a different length than this, it
+        will be a runtime SDE.
+
+        Everything from here to the end of this sequence is unmodified from
+        the original IPV4 schema it is based on.
+        -->
+        <xs:element name="Version" type="tns:bit" dfdl:length="4"/>
+        <xs:element name="IHL" type="tns:bit" dfdl:length="4"/>
+        <xs:element name="DSCP" type="tns:bit" dfdl:length="6"/>
+        <xs:element name="ECN" type="tns:bit" dfdl:length="2"/>
+        <xs:element name="Length" type="tns:bit" dfdl:length="16"/>
```

Comment:
Enhance this schema so that length depends on an element later after the checksum layer. 
This will exercise the ability to suspend while populating this part1 layer.